lib/frontend/src/frontend.rs

```diff
@@ -53,12 +54,16 @@ struct Position {
 }
 
 impl Position {
-    fn beginning() -> Self {
+    fn default() -> Self {
         Self {
-            ebb: Ebb::new(0),
-            basic_block: Block::new(0),
+            ebb: Ebb::reserved_value(),
+            basic_block: Block::reserved_value(),
         }
```

Comment:
Nice. That should get you manny assertions if you try to use it.

Reply:
Yeah, except I realized that this is essentially just reimplementing `PackedOption`. So I changed it to use `PackedOption` now.